munit/shared/src/main/scala/munit/Printer.scala

```diff
@@ -14,6 +14,84 @@ trait Printer {
   def height: Int = 100
   def isMultiline(string: String): Boolean =
     string.contains('\n')
+
+  /**
+   * Combine two printers into a single printer.
+   *
+   * Order is important : this printer will be tried first, then the other printer.
+   * The new Printer's height will be the max of the two printers' heights.
+   *
+   * Comibining two printers can be useful if you want to customize Printer for
+   * some types somewhere, but want to add more specialized printers for some tests
+   *
+   * {{{
+   * trait MySuites extends FunSuite {
+   *   override val printer = Printer.apply {
+   *     case long => s"${l}L"
+   *   }
+   * }
+   *
+   * trait SomeOtherSuites extends MySuites {
+   *   override val printer = Printer.apply {
+   *     case m: SomeCaseClass => m.someCustomToString
+   *   } orElse super.printer
+   * }
+   *
+   * }}}
+   */
+  def orElse(other: Printer): Printer = {
+    val h = this.height
+    val p: (Any, StringBuilder, Int) => Boolean = this.print
+    new Printer {
+      def print(value: Any, out: StringBuilder, indent: Int): Boolean =
+        p.apply(value, out, indent) || other.print(
+          value,
+          out,
+          indent
+        )
+      override def height: Int = h.max(other.height)
+    }
+  }
+
+}
+
+object Printer {
+
+  def apply(
+      height: Int
+  )(partialPrint: PartialFunction[Any, String]): Printer = {
+    val h = height
+    new Printer {
+      def print(value: Any, out: StringBuilder, indent: Int): Boolean =
+        value match {
+          case simpleValue =>
+            partialPrint.lift.apply(simpleValue).fold(false) { string =>
+              out.append(string)
+              true
+            }
+        }
```

Comment:
doesn't this match expression always match? What's the difference with:

```scala
def print(value: Any, out: StringBuilder, indent: Int): Boolean =
  partialPrint.lift.apply(value).fold(false) { string =>
    out.append(string)
    true
  }
```

Also, super minor preference, but I find `fold` not super straightforward to read compared to:

```scala
def print(value: Any, out: StringBuilder, indent: Int): Boolean =
  partialPrint.lift.apply(value) match {
    case Some(string) =>
    	out.append(string)
    	true
    case None => false
  }
```

Reply:
Damnit yeah it always match Must be a leftover from when I tried to include some recursivity and handle Arrays, Products....before realising it was a repeat of the logic which should be applied after the custom printer. Will remove it. Same for the fold !